compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -474,11 +474,14 @@ fn subst_and_check_impossible_predicates<'tcx>(
     result
 }
 
-/// Checks whether a trait's method is impossible to call on a given impl.
+/// Checks whether a trait's associated item is impossible to reference on a given impl.
 ///
 /// This only considers predicates that reference the impl's generics, and not
 /// those that reference the method's generics.
-fn is_impossible_method(tcx: TyCtxt<'_>, (impl_def_id, trait_item_def_id): (DefId, DefId)) -> bool {
+fn is_impossible_associated_item(
```

Comment:
I've renamed this since the previous name confused me quite a bit at first, I wasn't sure if the ICE happens due to the method w/ RPITIT or due to the synthetic RPITIT type. It can every kind of associated item just fine (even generic consts I assume) and rustdoc does use the query for every kind already.

Reply:
Yeah, it used to be used during codegen to skip monomorphizing methods that have impossible to solve predicates, but now it's used for other things.